src/dbnode/storage/series/series.go

```diff
@@ -592,8 +593,10 @@ func (s *dbSeries) OnEvictedFromWiredList(id ident.ID, blockStart time.Time) {
 	s.Lock()
 	defer s.Unlock()
 
-	// Should never happen
-	if !id.Equal(s.id) {
+	// If the series is closed, the ID will be nil, causing the ID equality
+	// check to panic. Further, closing the series resets the internal
+	// cached blocks, so we don't need to continue with the logic below.
+	if s.closed || !id.Equal(s.id) {
```

Comment:
Do we need `closed`? I think it might be better to do nil checks, e.g.:

```go
if id == nil || s.id == nil || !id.Equal(s.id) {
  return
}
```

Also, it's not clear whether we need to be worried about `s` itself being nil, but in that case:

```go
if s == nil || id == nil {
  return
}

s.Lock()
defer s.Unlock()

if s.id == nil || !id.Equal(s.id) {
  return
}
```

Reply:
Sounds good.